os/arch/arm/src/amebasmart/amebasmart_serial.c

```diff
@@ -1377,6 +1380,44 @@ int up_getc(void)
 	return ch;
 }
 
+/****************************************************************************
+ * Name: up_flush_console
+ *
+ * Description:
+ *    This function is used to ensure that all characters in the UART buffer
+ *    are transmitted.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+void up_flush_console(void)
+{
+#ifdef CONFIG_UART4_SERIAL_CONSOLE
+	uint16_t nbyte;
+	irqstate_t flags = enter_critical_section();
+
+	/* To avoid duplicated calling up_flush_console(). */
+	if (g_log_flush_running) {
+		leave_critical_section(flags);
+		return;
+	}
+	g_log_flush_running = true;
+
+	do {
+		while (!LOGUART_Ready());
+		nbyte = uart_xmitchars(&CONSOLE_DEV);
+	} while (nbyte);
```

Comment:
How does this work? Currently it looks like we still rely on uart interrupts to send data. But interrupts are disabled. Wont it be better to use lldbg path instead?

Reply:
The IRQ is calling uart_xmitchars()

Reply:
I meant uart_xmitchars path is through drivers (will call the drivers send function and it also has some signalling logic to wakeup other threads), but at this point, we are no longer using the driver's path to print logs. Instead we use lldbg, so I asked the above question, why not use up_putc instead of driver send.

Reply:
we need signalling logic for reloading